File: Maximize sum after K negations - GFG/maximize-sum-after-k-negations.cpp
//{ Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution{
    public:
    long long int maximizeSum(long long int a[], int n, int k)
    {
        // Your code goes here
         sort(a, a+n);
        long long int sum=0;
        for(int i=0; i<n; i++)
        {
            if(a[i]<0 && k>0)
            {
                a[i]=-a[i];
                k--;
            }
        }
        
        for(int i=0; i<n; i++)
        {
            sum+=a[i];
        }
        
        int x=*min_element(a, a+n);
        if(k&1)  sum-=2*x;
        
        return sum;
    }
};

//{ Driver Code Starts.
int main()
 {
     int t;
     cin>>t;
     while(t--)
     {
         int n, k;
         cin>>n>>k;
         long long int a[n+5];
         for(int i=0;i<n;i++)
         cin>>a[i];
         Solution ob;
         cout<<ob.maximizeSum(a, n, k)<<endl;
     }
	
	return 0;
}
// } Driver Code Ends

File: Maximize sum after K negations - GFG/maximize-sum-after-k-negations.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main(String[] args) throws IOException
	{
	        BufferedReader br =
            new BufferedReader(new InputStreamReader(System.in));
        int t =
            Integer.parseInt(br.readLine().trim()); // Inputting the testcases
        while(t-->0)
        {
            StringTokenizer stt = new StringTokenizer(br.readLine());
            
            int n = Integer.parseInt(stt.nextToken());
            int k = Integer.parseInt(stt.nextToken());
            long a[] = new long[(int)(n)];
            String inputLine[] = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                a[i] = Long.parseLong(inputLine[i]);
            }
            
            Solution obj = new Solution();
            System.out.println(obj.maximizeSum(a, n, k));
            
        }
	}
}

// } Driver Code Ends


//User function Template for Java


class Solution {

    public static long maximizeSum(long a[], int n, int k)
    {
        // Your code goes here  
        
         Arrays.sort(a);

         long sum = 0;
         
         for (int i = 0; i < n; i++) {
            if (a[i] < 0 && k > 0) {
              a[i] = -a[i];
              k--;
        }
        sum += a[i];
    }

         // Find the minimum element in the array
        long x = Arrays.stream(a).min().getAsLong();

        if (k % 2 == 1) {
           sum -= 2 * x;
        }

        return sum;
    }
}